### DISCUSSION TITLE
Inertia multi rootView config

### DISCUSSION THREAD

**Discussion by kitro** [adonisjs/core#4580] [Help] [answered]
I'm testing Adonis with Inertia adapter, I see that we can define the `rootView` in `confg/inertia.ts` file. is there are any possibility to define multi `rootView`, in Laravel we can do that by introduce separate middleware. I tried in Adonis to create a new named middleware that extends `InertiaMiddleware` but I'm stuck how I can define the config explicitly or override it to read the config from different location than `config/inertia.ts`.
Thanks for help!

**Comment by Julien-R44** [comment, marked as answer]
Hey, we don't have anything for that today. Having to add middleware for that seems a bit cumbersome to me

What about allowing a function for `rootView` in the `config/inertia.ts` file instead? 

```ts
import { defineConfig } from '@adonisjs/inertia';

export default defineConfig({
  rootView: ({ request }: HttpContext) => {
    if (request.url().startsWith('/admin')) {
      return 'admin_layout';
    }

    return 'inertia_layout';
  },
})
```

Happy to accept a PR for that

**Reply by kitro** [reply]
Thank you @Julien-R44 for the suggestion. I'm continue my experiments in Adonis.

**Reply by Julien-R44** [reply]
Now available with this release : https://github.com/adonisjs/inertia/releases/tag/v1.0.0-29
